File: python/0001_two_sum.py
# https://leetcode.com/problems/two-sum/
# O(n) time complexity
# O(n) space complexity
def two_sum(nums, target):
    dictionary = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in dictionary:
            return [dictionary[complement], i]
        dictionary[num] = i

File: javascript/0001_two_sum.js
// https://leetcode.com/problems/two-sum/
// O(n) time complexity
// O(n) space complexity
function twoSum(nums, target) {
  let mappedToIndex = new Map();
  for (const [idx, num] of nums.entries()) {
    const complement = target - num;
    if (mappedToIndex.has(complement)) {
      return [mappedToIndex.get(complement), idx];
    }
    mappedToIndex.set(num, idx);
  }
}
